mx_fetchjdk.py

```diff
@@ -0,0 +1,213 @@
+#
+# ----------------------------------------------------------------------------------------------------
+#
+# Copyright (c) 2020, Oracle and/or its affiliates. All rights reserved.
+# DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+#
+# This code is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 2 only, as
+# published by the Free Software Foundation.
+#
+# This code is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+# version 2 for more details (a copy is included in the LICENSE file that
+# accompanied this code).
+#
+# You should have received a copy of the GNU General Public License version
+# 2 along with this work; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+# or visit www.oracle.com if you need additional information or have any
+# questions.
+#
+# ----------------------------------------------------------------------------------------------------
+#
+from __future__ import print_function
+import os, shutil, json
+from os.path import join, exists, abspath, isdir, islink
+from shutil import copytree
+from argparse import ArgumentParser, REMAINDER
+
+from mx import optional_suite_context, _mx_home, command
+from jdk_distribution_parser import JdkDistribution
+import mx, mx_urlrewrites
+
+@command('mx', 'fetch-jdk', '[options]')
+@optional_suite_context
+def fetch_jdk(args):
+    """fetches required JDK version
+
+    If called without --quiet flag, menu will be printed for available JDK selection.
+    """
+    args = _parse_fetchjdk_settings(args)
+
+    distribution = args["java-distribution"]
+    jdk_path = args["jdk-path"]
+    jdk_artifact = distribution.get_jdk_folder()
+    full_jdk_path = distribution.get_full_jdk_path(jdk_path)
+    jdk_url = mx_urlrewrites.rewriteurl(distribution.get_url())
+    jdk_url_sha = jdk_url + ".sha1"
+    jdk_archive = distribution.get_archive()
+
+    if not args["quiet"] and not mx.ask_yes_no("Install {} to {}".format(jdk_artifact, jdk_path), default='y'):
+        mx.abort("JDK installation canceled")
+
+    if not exists(full_jdk_path):
+        if not args["quiet"]:
+            print("Fetching {} archive...".format(jdk_artifact))
+
+        archive_location = join(jdk_path, jdk_archive)
+        mx._opts.no_download_progress = args["quiet"]
+        try:
+            sha1_hash = mx._hashFromUrl(jdk_url_sha)
```

Comment:
Seems like `sha1_hash` is of type `bytes` which causes a problem when running with Python3:
```
[1]  openjdk8                  | 8u252+09-jvmci-20.1-b02
[2]  openjdk8-fastdebug        | 8u252+09-jvmci-20.1-b02
[3]* labsjdk-ce-11             | ce-11.0.7+10-jvmci-20.1-b02
[4]  labsjdk-ce-11-debug       | ce-11.0.7+10-jvmci-20.1-b02
Select JDK>
Install labsjdk-ce-11-jvmci-20.1-b02 to /Users/dnsimon/mx/bin/jdks? [Yn]:
Fetching labsjdk-ce-11-jvmci-20.1-b02 archive...
Traceback (most recent call last):
  File "/Users/dnsimon/mx/mx.py", line 19529, in <module>
    main()
  File "/Users/dnsimon/mx/mx.py", line 19510, in main
    retcode = c(command_args)
  File "/Users/dnsimon/mx/mx_commands.py", line 147, in __call__
    return self.command_function(*args, **kwargs)
  File "/Users/dnsimon/mx/mx_fetchjdk.py", line 69, in fetch_jdk
    mx.download_file_with_sha1(jdk_artifact, archive_location, [jdk_url], sha1_hash, archive_location + '.sha1', resolve=True, mustExist=True, sources=False)
  File "/Users/dnsimon/mx/mx.py", line 3499, in download_file_with_sha1
    cachePath = _get_path_in_cache(name, sha1, urls, sources=sources)
  File "/Users/dnsimon/mx/mx.py", line 168, in _get_path_in_cache
    return join(_cache_dir(), name + '_' + sha1 + ('.dir' if not ext else ''), filename)
TypeError: can only concatenate str (not "bytes") to str
```

Reply:
Fixed